src/resolve/resolved-dns-stub.c

```diff
@@ -528,6 +603,73 @@ static int manager_dns_stub_tcp_fd(Manager *m) {
         return m->dns_stub_tcp_fd = TAKE_FD(fd);
 }
 
+static int manager_dns_stub_tcp_fd_extra(Manager *m, DNSStubListenerExtra *l) {
+        _cleanup_free_ char *pretty = NULL;
+        _cleanup_close_ int fd = -1;
+        int r;
+
+        if (l->fd >= 0)
+                return 0;
+
+        fd = socket(socket_address_family(&l->address), SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0);
+        if (fd < 0) {
+                r = -errno;
+                goto fail;
+        }
+
+        r = set_dns_stub_common_socket_options(fd);
+        if (r < 0)
+                goto fail;
+
+        r = setsockopt_int(fd, IPPROTO_IP, IP_TTL, true);
+        if (r < 0)
+                goto fail;
```

Comment:
hmm, this doesn't look right. First of all this sockopt is actually not a boolean, but takes arbitrary integers. This is a pre-existing bug where ths was copied from. i prepped #16946 fixing that tree wide.

But I think this shouldn't be done for the extra stub listeners at all. That's because this is suppsoed to make sure that DNS traffic we generate on the local stub doesn't ever get routed, because the ttl doesn't allow it. It's a safety precaution. But my guess is that when people use the extra stub listeners, they probably put them on non-local ipaddresses, and when they do they probably want ttls > 1. Hence: just drop these three lines altogether, please